frappe/modules/full_text_search.py

```diff
@@ -10,97 +10,219 @@
+			parser.remove_plugin_class(WildcardPlugin)
+			query = parser.parse(text)
+
+			filter_scoped = None
+			if scope:
+				filter_scoped = Prefix("path", scope)
+			results = searcher.search(query, limit=limit, filter=filter_scoped)
+
+			for r in results:
+				title_highlights = r.highlights("title")
+				content_highlights = r.highlights("content")
+				out.append(
+					frappe._dict(
+						title=r["title"],
+						path=r["path"],
+						title_highlights=title_highlights,
+						content_highlights=content_highlights,
+					)
+				)
 
-def build_index_for_all_routes():
-	print("Building search index for all web routes...")
-	routes = get_routes_to_index()
-	documents = [get_document_to_index(route) for route in routes]
-	build_index("web_routes", documents)
+		return out
 
 
-@frappe.whitelist(allow_guest=True)
-def web_search(index_name, query, scope=None, limit=20):
-	limit = cint(limit)
-	return search(index_name, query, scope, limit)
+def get_doctype_routes_with_web_view():
+	all_routes = []
+	filters = { "has_web_view": 1, "allow_guest_to_view": 1 }
+	fields = ["name", "is_published_field"]
+	doctype_with_web_views = frappe.get_all("DocType", filters=filters, fields=fields)
 
+	for doctype in doctype_with_web_views:
+		if doctype.is_published_field:
+			routes = frappe.get_all(doctype.name, filters={doctype.is_published_field: 1}, fields="route")
+			all_routes += [route.route for route in routes]
 
-def get_document_to_index(route):
-	frappe.set_user("Guest")
-	frappe.local.no_cache = True
+	return all_routes
 
-	try:
-		set_request(method="GET", path=route)
-		content = render_page(route)
-		soup = BeautifulSoup(content, "html.parser")
-		page_content = soup.find(class_="page_content")
-		text_content = page_content.text if page_content else ""
-		title = soup.title.text.strip() if soup.title else route
+def get_static_pages_from_all_apps():
+	apps = frappe.get_installed_apps()
 
-		frappe.set_user("Administrator")
+	routes_to_index = []
+	for app in apps:
+		base = frappe.get_app_path(app, 'www')
+		path_to_index = frappe.get_app_path(app, 'www')
 
-		return frappe._dict(title=title, content=text_content, path=route)
-	except (
-		frappe.PermissionError,
-		frappe.DoesNotExistError,
-		frappe.ValidationError,
-		Exception,
-	):
-		pass
+		for dirpath, _, filenames in os.walk(path_to_index, topdown=True):
+			for f in filenames:
+				if f.endswith(('.md', '.html')):
+					filepath = os.path.join(dirpath, f)
 
+					route = os.path.relpath(filepath, base)
+					route = route.split('.')[0]
 
-def build_index(index_name, documents):
-	schema = Schema(
-		title=TEXT(stored=True), path=ID(stored=True), content=TEXT(stored=True)
-	)
+					if route.endswith('index'):
+						route = route.rsplit('index', 1)[0]
 
-	index_dir = get_index_path(index_name)
-	frappe.create_folder(index_dir)
+					routes_to_index.append(route)
 
-	ix = create_in(index_dir, schema)
-	writer = ix.writer()
+	return routes_to_index
```

Comment:
how about something like:

```py
from glob import glob

routes_to_index = []
for app in apps:
    path_to_index = frappe.get_app_path(app, 'www')

    files_to_index = glob(path_to_index + '/**/*.html', recursive=True)
    files_to_index.extend(glob(path_to_index + '/**/*.md', recursive=True))
    for file in files_to_index:
        route = os.path.relpath(file, path_to_index).split('.')[0]
        if route.endswith('index'):
            route = route.rsplit('index', 1)[0]
        routes_to_index.append(route)
return routes_to_index
```

lesser loops, lesser variables, faster, same functionality.

Reply:
It's not necessarily faster, but it's better nonetheless. Less prone to error than the previous implementation.